server/render.js

```diff
@@ -132,3 +136,36 @@ function errorToJSON (err) {
 
   return json
 }
+
+export async function serveStaticWithGzip (req, res, path) {
+  const encoding = accepts(req).encodings(['gzip'])
+  if (encoding !== 'gzip') {
+    return serveStatic(req, res, path)
+  }
+
+  const gzipPath = `${path}.gz`
+  const exists = await fs.exists(gzipPath)
+  if (!exists) {
+    return serveStatic(req, res, path)
+  }
+
+  res.setHeader('Content-Encoding', 'gzip')
+  return serveStatic(req, res, gzipPath)
+}
+
+export function serveStatic (req, res, path) {
+  return new Promise((resolve, reject) => {
+    send(req, path)
+    .on('error', (err) => {
+      if (err.code === 'ENOENT') {
+        res.statusCode = 404
+        res.end('Not Found')
```

Comment:
In the orginal 404, it sends a bunch of HTML too. It's okay for pages, but this is just a JSON endpoint. We don't need that. (Since user won't access these files directly)